mobile/lib/entities/album.entity.dart

```diff
@@ -114,7 +114,6 @@ class Album {
         ) &&
         shared == other.shared &&
         activityEnabled == other.activityEnabled &&
-        owner.value == other.owner.value &&
```

Comment:
Why is this removed?
Do we instead need a special method to compare albums regardless of the owner?

Reply:
From my observation, the `IsarLink` here changes the hashcode of the object even if the value is the same. So when we create a `Set` of `Album`, they aren't correctly compared.

https://github.com/immich-app/immich/blob/5e8616b45cd8497dd52d1a2eeee5149c6f871738/mobile/lib/services/album.service.dart#L173

Reply:
since this specific case is just server albums and we don't want changes to asset.entity.dart break this: we can just specify to use the remoteId for the set operations (dart sets allow to use custom hash/equals operations)

Reply:
Do you know the syntax for this?

Reply:
> Do you know the syntax for this?

You can probably use a [SplayTreeSet](https://api.flutter.dev/flutter/dart-collection/SplayTreeSet-class.html)

Reply:
Far simpler:
https://api.flutter.dev/flutter/dart-collection/HashSet/HashSet.html
You can just set a custom compare and hash code operation in the set constructor

Reply:
Thanks @fyfrey for a cool knowledge. The HashSet is now implemented. Please give it another round of review and approve if all look good